src/keymap.rs

```diff
@@ -13,6 +14,58 @@ use crate::tty::{RawReader, Term, Terminal};
 /// The number of times one command should be repeated.
 pub type RepeatCount = usize;
 
+/// Custom dynamic action
+#[derive(Clone)]
+pub struct Action {
+    pub name: String,
+    /// Takes the currently edited `line` with the cursor `pos`ition and
+    /// returns the command to be performed or `None` to perform the default
+    /// one.
+    pub action: fn(line: &str, pos: usize, ctx: &ActionContext) -> Option<Cmd>,
```

Comment:
I don't understand why `Action::action` contains a function. Rather than `Action` being a trait.

Reply:
Mainly because `Action` is `Sized`.

Reply:
`Box<dyn Action>` is also sized. And I don't think it can be a performance issue here.